File: models/Assessment.js
const mongoose = require('mongoose');

const assessmentSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  openness: {
    type: Number,
    required: true,
  },
  conscientiousness: {
    type: Number,
    required: true,
  },
  extraversion: {
    type: Number,
    required: true,
  },
  agreeableness: {
    type: Number,
    required: true,
  },
  neuroticism: {
    type: Number,
    required: true,
  },
  comments: {
    type: String,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

const Assessment = mongoose.model('Assessment', assessmentSchema);

module.exports = Assessment;

File: OceanPersonalityApi/Models/Assessment.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace OceanPersonalityApi.Models
{
    public class Assessment
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }

        [BsonRepresentation(BsonType.ObjectId)]
        public string UserId { get; set; } = null!;

        public double Openness { get; set; }

        public double Conscientiousness { get; set; }

        public double Extraversion { get; set; }

        public double Agreeableness { get; set; }

        public double Neuroticism { get; set; }

        public string? Comments { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}